x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/integrations/hooks.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { IntegrationCardItem } from '@kbn/fleet-plugin/public';
+
+import { useNavigation } from '../../../../../common/lib/kibana';
+import { getFilteredCards } from './utils';
+import { INTEGRATION_TABS } from './const';
+
+export const useIntegrationCardList = ({
+  integrationsList,
+  customCardNames,
+}: {
+  integrationsList: IntegrationCardItem[];
+  customCardNames?: string[] | undefined;
+}): IntegrationCardItem[] => {
+  const { navigateTo, getAppUrl } = useNavigation();
+  const { featuredCards, integrationCards } = useMemo(
+    () => getFilteredCards({ navigateTo, getAppUrl, integrationsList, customCardNames }),
+    [navigateTo, getAppUrl, integrationsList, customCardNames]
+  );
+
+  if (customCardNames && customCardNames.length > 0) {
+    return Object.values(featuredCards) ?? [];
+  }
+  return integrationCards ?? [];
+};
+
+export const useTabMetaData = (toggleIdSelected: string) => {
```

Comment:
Do we really need to extract this hook? We know the tab will always exist. Assuming we create the `IntegrationTabId` enum, we can map all tabs in the constants file with:
```
export const INTEGRATION_TABS_BY_ID = Object.fromEntries(
  INTEGRATION_TABS.map((tab) => [tab.id, tab])
) as Record<IntegrationTabId, Tab>;
```
And then we can do this in-place:
```
const selectedTab = useMemo(
    () => INTEGRATION_TABS_BY_ID[selectedTabId],
    [selectedTabId]
  );
```
So `selectedTab` can't be `undefined`.